src/v2/options.ts

```diff
@@ -182,6 +188,11 @@ export interface GlobalOptions {
    * User labels to set on the function.
    */
   labels?: Record<string, string>;
+
+  /**
+   * Invoker to set access control on https functions.
+   */
+  invoker?: Invoker | Invoker[];
```

Comment:
Given your new defensive programming stance I agree with Daniel now that we can/should change our typing.

```
invoker?: 'public' | 'private' | string | string[];
```

This will make sure code complete doesn't suggest "public" or "private" when the customer enters an array.

Reply:
Changed the type definition